include/dali/core/error_handling.h

```diff
@@ -234,28 +248,28 @@ inline dali::string GetStacktrace() {
 #define FILE_AND_LINE __FILE__ ":" DALI_STR(__LINE__)
 
 #define DALI_MESSAGE_WITH_STACKTRACE(str)\
-  (std::string("[" FILE_AND_LINE "] ") + str + dali::GetStacktrace())
+  (std::string("[" FILE_AND_LINE "] ")), (std::string(str)), dali::GetStacktrace()
```

Comment:
```suggestion
#define DALI_MESSAGE_WITH_STACKTRACE(...)\
  ("[" FILE_AND_LINE "] "), ##__VA_ARGS __, dali::GetStacktrace()
```
This way you allow the message to be empty, to be multi-part and of heterogeneous types.
Also, you save the unnecessary construction of an `std::string` out of a string literal.

Reply:
done